client/data/marketplace/use-wpcom-plugins-query.ts

```diff
@@ -89,14 +90,16 @@ const fetchWPCOMPlugin = ( slug: string ) =>
 		apiNamespace: pluginsApiNamespace,
 	} );
 
-export const getWPCOMPluginQueryParams = ( slug: string ): [ QueryKey, QueryFunction ] => {
-	const cacheKey = getCacheKey( slug + '-normalized' );
-	const fetchFn = () =>
+export const getWPCOMPluginQueryParams = (
+	slug: string
+): { queryKey: QueryKey; queryFn: QueryFunction } => {
+	const queryKey = getCacheKey( slug + '-normalized' );
+	const queryFn = () =>
 		fetchWPCOMPlugin( slug ).then( ( data: any ) =>
 			normalizePluginData( { detailsFetched: Date.now() }, data )
 		);
 
-	return [ cacheKey, fetchFn ];
+	return { queryKey, queryFn };
```

Comment:
Noting that `getWPCOMPluginQueryParams` is used here:

https://github.com/Automattic/wp-calypso/blob/8f1fa912356c59738ed483a536f84bb85f191ca1/client/my-sites/plugins/controller-logged-out.js#L78

and we might need to do some changes to accommodate the function signature difference.